### PR TITLE
[#243] Mat Resource View (connects #243 )

### PR DESCRIPTION
**Pull request by Pr0chin**

### CODE REVIEW

**Comment by paulbert** [comment]
Made some changes because there was a big chunk of copied code in `style.scss` to simply separate them from the `.list-item` parent which could just as easily be applied to the `mat-row` element.  Please do not duplicate code like this.  If you are having trouble finding a different solution, you can ask someone for help in the PR or on Gitter.

In addition I went through and simplified some of the styles and fixed the delete function so the deleted resource would be removed from the list.

File: src/app/resources/resources.component.html [hunk 1]
```diff
@@ -1,52 +1,56 @@
...
-      <h3>{{ item.doc.title }}</h3>
+<div>
+  <button mat-raised-button routerLink="add" color="primary" routerLinkActive="active" i18n>Add Resource</button>
+</div>
+<div>
+  <mat-form-field>
+    <input matInput (keyup)="applyResFilter($event.target.value)" i18n-placeholder placeholder="Search">
+  </mat-form-field>
+</div>
+<mat-table #table [dataSource]="resources" matSort>
```
Comment (nirojdyola) [inline src/app/resources/resources.component.html]: matSort is not need. Could you please remove this ?

File: src/app/resources/resources.component.html [hunk 2]
```diff
@@ -1,52 +1,56 @@
...
-          <div [ngStyle]="{ 'background-color': 'gold', 'height': '8px', 'width': fRating + '%'}"></div>
+        <div class="label">
+          <span class="small">Rating ({{ rating }})</span>
+          <span class="small">Female ({{ fRating }}%)</span>
+          <span class="small">Male ({{ mRating }}%)</span>
         </div>
-        <div class="rating-bar">
-          <div [ngStyle]="{ 'background-color': 'gold', 'height': '8px', 'width': mRating + '%'}"></div>
+        <div class="info">
+          <span> {{ rating }}
```
Comment (nirojdyola) [inline src/app/resources/resources.component.html]: should be in a single line. Closing tag of span is in another line , that needs to be fix. Please check it once .

File: src/app/resources/resources.component.ts [hunk 3]
```diff
@@ -63,7 +74,7 @@ export class ResourcesComponent implements OnInit {
       const { _id: resourceId, _rev: resourceRev } = resource;
       this.couchService.delete('resources/' + resourceId + '?rev=' + resourceRev)
```
Comment (nirojdyola) [inline src/app/resources/resources.component.ts]: 'resources' can be change to this.dbName

File: src/app/resources/resources.component.html [hunk 4]
```diff
@@ -1,52 +1,58 @@
-<!-- FOR DEMO ONLY -->
-<!-- use the async pipe when fetching for database -->
-<button mat-raised-button routerLink="add" color="primary" routerLinkActive="active" i18n>Add Resource</button>
-<ng-template ngFor let-item [ngForOf]="resources">
-  <!-- Used ng-template insted of spans or divs because I did not want any to add any element -->
-  <div class="list-item">
-    <div class="list-item-info">
-      <h3>{{ item.doc.title }}</h3>
+<div>
+  <button mat-raised-button routerLink="add" color="primary" routerLinkActive="active" i18n>Add Resource</button>
```
Comment (lmmrssa) [inline src/app/resources/resources.component.html]: routerlinkactive not required

**Pull request merged**